compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1209,6 +1307,8 @@ impl<'context> Elaborator<'context> {
         }
 
         for ((self_type, local_module), function_sets) in impls {
+            self.push_scope();
```

Comment:
We need to push a scope here as to not have duplicate definition errors when there is a global with the same name as the numeric generic in the type definition.